traffic_stats/assert/testhelpers.go

```diff
@@ -0,0 +1,109 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package assert
+
+import (
+	"reflect"
+	"testing"
+)
+
+// NotNil takes a value, and checks whether that value is not nil, failing the test
+// if not
+func NotNil(t *testing.T, a interface{}) {
+	if a == nil {
+		t.Errorf("expected %v to be non-nil, but it is", a)
+	}
+}
+
+// Nil takes a value, and checks whether that value is nil, failing the test
+// if not
+func Nil(t *testing.T, a interface{}) {
+	if a != nil {
+		t.Errorf("expected %v to be nil, but it's not", a)
+	}
+}
+
+// Equal takes two values, checks whether they are equal (using a deep equal
+// check) and fails the test if not
+func Equal(t *testing.T, a, b interface{}) {
+	if !reflect.DeepEqual(a, b) {
```

Comment:
DeepEqual can be slow, and might be unexpected, callers might expect 'Equal' to actually check '=='. Might be worth making a separate `func Equal` and `func DeepEqual`